src/main/java/com/hillel/dzhurynskyi/homework/hm6/ArrayValueCalculator.java

```diff
@@ -0,0 +1,29 @@
+package com.hillel.dzhurynskyi.homework.hm6;
+
+public class ArrayValueCalculator {
+
+    public int doCalc(String [][]arr) throws ArraySizeException, ArrayDataException{
+        int suma = 0;
+
+        if (arr.length > 4) {
+            throw new ArraySizeException();
+        } else {
+            for (int i = 0; i < 4; i++) {
+                if (arr[i].length > 4) {
+                    throw new ArraySizeException("index "+i);
+                } else {
+                    for (int j = 0; j < 4; j++) {
+                        System.out.print(arr[i][j] + " ");
+                        try {
+                            suma += Integer.parseInt(arr[i][j]);
+                        } catch (NumberFormatException e) {
+                            throw new ArrayDataException(i, j, arr[i][j]);
```

Comment:
In your solution possible to throw ArrayDataException for an invalid array size (ArraySizeException should be thrown because the array size is invalid, but we check before cells format). Example
String[][] arr = {
{"1", "XXX", "567", "1234"},
{"45"},
{"45", "998", "567", "1234"},
{"45", "998", "567", "456"}
};